### ISSUE TITLE
Add post-action callbacks

### ISSUE THREAD

**Issue by mwvaughn** [TACC/abaco#79]
I'm noticing that users are adopting increasing amounts of orchestration as they build and deploy and use Abaco. A limitation of the current system is that one must poll for completion of a given task (deploy, update, execute, scale up, delete, etc). The obvious exception is when one is using the upcoming synchronous endpoint, but that doesn't cover the other use cases anyway. Further validation of the possible need for this is that where we have implemented callback support in the client-side SDK, it has been a valuable addition to the orchestration toolkit. I propose at least adding at least a simple post-action callback mechanism to Abaco. We can discuss the implementation details below, but I wanted to get this pinned to the board for consideration.

**Comment by joestubbs** [comment]
I agree completely, let's hammer out some initial use cases. I've been doing some research on event systems in support of Tapis v3 any way. I think Abaco can serve as an early adopter and provide use cases, requirements, and even implement a prototype, depending on our time table and delivery schedule.
